demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -546,6 +521,410 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+                      , cv::GMat
+                      , FaceRecognizerConfig)>
+            , "custom.get_recognize_result") {
+        static std::tuple < cv::GArrayDesc
+                          , cv::GArrayDesc
+                          , cv::GArrayDesc> outMeta( const cv::GArrayDesc &
+                                                   , const cv::GArrayDesc &
+                                                   , const cv::GArrayDesc &
+                                                   , const cv::GMatDesc &
+                                                   , const FaceRecognizerConfig &) {
+            return std::make_tuple( cv::empty_array_desc()
+                                  , cv::empty_array_desc()
+                                  , cv::empty_array_desc());
+        }
+    };
+
+    G_API_OP( GetActionTopHandsDetectionResult
+            , <cv::GArray<TrackedObject>( cv::GArray<DetectedAction>
+                                        , cv::GMat)>
+            , "custom.get_action_detection_result_for_top_k_first_hands") {
+        static cv::GArrayDesc outMeta( const cv::GArrayDesc &
+                                     , const cv::GMatDesc &) {
+            return cv::empty_array_desc();
+        }
+    };
+
+    G_API_OP( AlignFacesForReidentification
+            , <cv::GMat( cv::GArray<cv::GMat>
+                       , cv::GArray<cv::Rect>
+                       , cv::GMat)>
+            , "custom.align_faces_for_reidentification") {
+        static cv::GMatDesc outMeta(const cv::GArrayDesc &,
+            const cv::GArrayDesc &,
+            const cv::GMatDesc & in) {
+            return in;
+        }
+    };
+
+    /** NOTE: Three crutch-Ops for empty {} GArray **/
+    G_API_OP( GetEmptyFaces
+            , <cv::GArray<detection::DetectedObject>(cv::GMat)>
+            , "custom.get_empty_faces_garray") {
+        static cv::GArrayDesc outMeta(const cv::GMatDesc &) {
+            return cv::empty_array_desc();
+        }
+    };
+
+    G_API_OP( GetEmptyActions
+            , <cv::GArray<DetectedAction>(cv::GMat)>
+            , "custom.get_empty_actions_garray") {
+        static cv::GArrayDesc outMeta(const cv::GMatDesc &) {
+            return cv::empty_array_desc();
+        }
+    };
+
+    G_API_OP( GetEmptyMatGArray
+            , <cv::GArray<cv::GMat>(cv::GMat)>
+            , "custom.get_empty_garray_of_mat") {
+        static cv::GArrayDesc outMeta(const cv::GMatDesc &) {
+            return cv::empty_array_desc();
+        }
+    };
+
+    /** OCV KERNELs */
+    GAPI_OCV_KERNEL(OCVFaceDetectorPostProc, FaceDetectorPostProc) {
+        static void run(const cv::Mat &in_ssd_result,
+                        const cv::Mat &in_frame,
+                        const detection::DetectorConfig & face_config,
+                        std::vector<detection::DetectedObject> &out_faces) {
+            std::unique_ptr<detection::FaceDetection> face_detector(new detection::FaceDetection(face_config));
```

Comment:
What's the reason to create it dynamically?

Why don't create it on stack?

If it has some complex initialization, could it be turned to a stateful kernel?

Reply:
Fixed.